### PR TITLE
[16.0][MIG] l10n_de_tax_statement

### PR DESCRIPTION
**Pull request by MrGigSolutions**
Updates for 16.0
- Moved the Update button to the top bar, and made the text that indicates to press this button an info text. The new layouting makes the button look out of place in the form
- Adjusted the tests to deal with the adjustments to accounting: journal_id has an invisible field in the form, so cannot be updated in the form, and account_id cannot be directly set. This was worked around by passing a default_journal_id, and by setting the correct default_account_id on this journal
- Adjusted translations file. Note: there are quite some fields in it that have no translation.

### CODE REVIEW

**Comment by MrGigSolutions** [comment]
Just like in the other module, the failing tests seem to be because of the 4 lines of demo data.

**Comment by MrGigSolutions** [comment]
@ortlam, would you mind having a look at this one?

**Comment by StefanRijnhart** [comment]
@hbrunn do these test failures make any sense to you? Asking because you migrated the module.

```
2024-01-22 06:26:49,695 250 ERROR odoo odoo.addons.datev_import_csv_dtvf.tests.test_datev_import_csv_dtvf: FAIL: TestDatevImportCsvDtvf.test_wizard_comma_separated
Traceback (most recent call last):
  File "/__w/l10n-germany/l10n-germany/datev_import_csv_dtvf/tests/test_datev_import_csv_dtvf.py", line 135, in test_wizard_comma_separated
    self._test_wizard("datev_import_csv_dtvf/examples/datev_export.csv")
  File "/__w/l10n-germany/l10n-germany/datev_import_csv_dtvf/tests/test_datev_import_csv_dtvf.py", line 123, in _test_wizard
    self.assertEqual(len(move.line_ids), 196)
AssertionError: 200 != 196
 
2024-01-22 06:26:49,697 250 INFO odoo odoo.addons.datev_import_csv_dtvf.tests.test_datev_import_csv_dtvf: Starting TestDatevImportCsvDtvf.test_wizard_semicolon_separated ... 
2024-01-22 06:26:50,224 250 WARNING odoo odoo.addons.datev_import_csv_dtvf.wizard.import_move: Approximate match: import account 7095 has been matched with Odoo account 7095000 
2024-01-22 06:26:51,081 250 INFO odoo odoo.addons.datev_import_csv_dtvf.wizard.import_move: Account moves IDs [32] created via file import 
2024-01-22 06:26:51,180 250 INFO odoo odoo.addons.datev_import_csv_dtvf.tests.test_datev_import_csv_dtvf: ====================================================================== 
2024-01-22 06:26:51,180 250 ERROR odoo odoo.addons.datev_import_csv_dtvf.tests.test_datev_import_csv_dtvf: FAIL: TestDatevImportCsvDtvf.test_wizard_semicolon_separated
Traceback (most recent call last):
  File "/__w/l10n-germany/l10n-germany/datev_import_csv_dtvf/tests/test_datev_import_csv_dtvf.py", line 138, in test_wizard_semicolon_separated
    self._test_wizard("datev_import_csv_dtvf/examples/datev_export_semicolon.csv")
  File "/__w/l10n-germany/l10n-germany/datev_import_csv_dtvf/tests/test_datev_import_csv_dtvf.py", line 123, in _test_wizard
    self.assertEqual(len(move.line_ids), 196)
AssertionError: 200 != 196
```

**Comment by hbrunn** [comment]
not much, but 196 is the correct value. would have to debug to find out where the extra move lines come from

**Comment by StefanRijnhart** [comment]
@MrGigSolutions see here: https://github.com/OCA/l10n-germany/pull/148

**Comment by MrGigSolutions** [comment]
retest

File: l10n_de_tax_statement/views/l10n_de_tax_statement_view.xml [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+                            name="post"
+                            string="Post"
+                            states="draft"
+                            type="object"
+                            class="oe_stat_button"
+                            icon="fa-arrow-right text-success"
+                        />
+                        <button
+                            name="reset"
+                            string="To Draft"
```
Comment (StefanRijnhart) [inline l10n_de_tax_statement/views/l10n_de_tax_statement_view.xml]: You changed this from 'Reset to Draft' to 'To Draft'. The `Reset` terminology is something of a convention of old to indicate the exceptional character of the state change. It is still used in Odoo on the account.move form, the payment form, and also in the NL counterpart of this module: https://github.com/OCA/l10n-netherlands/blob/16.0/l10n_nl_tax_statement/views/l10n_nl_vat_statement_view.xml#L36. Is the terminology change something of a personal preference or driven by functional considerations?

Comment (MrGigSolutions) [reply l10n_de_tax_statement/views/l10n_de_tax_statement_view.xml]: It looked too long. But I was not aware of the convention, so I'll change that back.

Comment (StefanRijnhart) [reply l10n_de_tax_statement/views/l10n_de_tax_statement_view.xml]: Thanks! Please squash it into the Migration to 16.0 commit.

**Review by StefanRijnhart — APPROVED** [review]

**Pull request closed without merging**